pkg/controller/che/che_cr_validator.go

```diff
@@ -23,7 +23,15 @@ import (
 // - self-contradictory configurations
 // - configurations with which it is impossible to deploy Che
 func ValidateCheCR(checluster *orgv1.CheCluster, isOpenshift bool) error {
-	if !isOpenshift {
+	if isOpenshift {
+		exposureStrategy := checluster.Spec.Server.ServerExposureStrategy
+		exposureType := checluster.Spec.Server.SingleHostExposureType
+		if exposureStrategy != "" && exposureStrategy != "multi-host" {
+			if exposureType != "" && exposureType != "gateway" {
```

Comment:
I assume you would not need this test and possible failure at runtime, if the `singleHostExposureType` was a K8S only field, just ignored for installations on OpenShift.

Reply:
Yep, I moved `singleHostExposureType` under K8s and removed this validation logic.